packages/amplify-prompts/src/progressbars/multibar.ts

```diff
@@ -0,0 +1,240 @@
+/**
+ * This class has been made as generic as possible to suit all use cases.
+ * But it is not without influence from the nuances of CloudFormation.
+*/
+import chalk from 'chalk';
+import { AmplifyTerminal as Terminal } from './terminal';
+import {
+  ProgressBar as Bar,
+  BarOptions,
+  ItemPayload,
+  ProgressPayload,
+} from './progressbar';
+
+/**
+ * Abstraction to create multiple progress bars inside one re writable block.
+ */
+export class MultiProgressBar {
+    private count : number;
+    private terminal : Terminal;
+    private options : BarOptions;
+    private bars : {name: string, bar: Bar}[];
+    private lastDrawnTime: number;
+    isActive: boolean;
+    private refreshRate: number;
+    private frameCount : number;
+    private frames : string[];
+    private timer!: ReturnType<typeof setTimeout>;
+    private prefixText : string;
+    private updated: boolean;
+    private lastDrawnStrings : string[];
+
+    constructor(options : BarOptions) {
+      this.terminal = new Terminal();
+      this.options = options;
+      this.bars = [];
+
+      if (this.options.hideCursor === true) {
+        this.terminal.cursor(false);
+      }
+
+      this.lastDrawnTime = Date.now();
+      this.isActive = false;
+      this.count = 0;
+
+      this.frameCount = 0;
+
+      // simulate spinner
+      this.frames = ['⠋', '⠙', '⠹', '⠸', '⠼', '⠴', '⠦', '⠧', '⠇', '⠏'];
+      this.prefixText = options.prefixText;
+      this.refreshRate = 200;
+
+      // Tracks when an update into one of the the bars happens
+      this.updated = false;
+      this.lastDrawnStrings = [];
+    }
+
+    /**
+     * Checks if the environment is tty
+     */
+    isTTY() : boolean {
```

Comment:
if possible, I'd recommend this method be private as it violates [tell don't ask](https://martinfowler.com/bliki/TellDontAsk.html)